packages/contractkit/src/wrappers/MetaTransactionWallet.ts

```diff
@@ -0,0 +1,279 @@
+import { ensureLeading0x, trimLeading0x } from '@celo/base'
+import { Signature } from '@celo/base/lib/signatureUtils'
+import { EIP712TypedData, generateTypedDataHash } from '@celo/utils/lib/sign-typed-data-utils'
+import { parseSignatureWithoutPrefix } from '@celo/utils/lib/signatureUtils'
+import BigNumber from 'bignumber.js'
+import { HttpProvider } from 'web3-core'
+import { Address } from '../base'
+import { MetaTransactionWallet } from '../generated/MetaTransactionWallet'
+import {
+  BaseWrapper,
+  CeloTransactionObject,
+  numericToHex,
+  proxyCall,
+  proxySend,
+  stringIdentity,
+  toTransactionObject,
+  valueToInt,
+} from './BaseWrapper'
+
+export interface MTWTransaction {
+  destination: Address
+  value?: BigNumber.Value
+  data?: string
+}
+
+export interface MTWMetaTransaction extends MTWTransaction {
+  nonce: number
+}
+
+export interface MTWSignedMetaTransaction extends MTWTransaction {
+  nonce: number
+  signature: Signature
+}
+
+/**
+ * Class that wraps the MetaTransactionWallet
+ */
+export class MetaTransactionWalletWrapper extends BaseWrapper<MetaTransactionWallet> {
+  /**
+   * Execute a transaction originating from the MTW
+   * Reverts if the caller is not a signer
+   * @param tx a MTWTransaction
+   */
+  executeTransaction(tx: MTWTransaction) {
+    return toTransactionObject(
+      this.kit,
+      this.contract.methods.executeTransaction(
+        tx.destination,
+        numericToHex(tx.value),
+        tx.data || '0x'
+      )
+    )
+  }
+
+  /**
+   * Execute a batch of transactions originating from the MTW
+   * Reverts if the caller is not a signer
+   * @param txs An array of MTWTransactions
+   */
+  executeTransactions(txs: MTWTransaction[]) {
+    return toTransactionObject(this.kit, this.buildExecuteTransactionsTx(txs))
+  }
+
+  /**
+   * Builds the TransactionObject for executeTransactions
+   * Reverts if the caller is not a signer
+   * @param txs An array of MTWTransactions
+   */
+  buildExecuteTransactionsTx(txs: MTWTransaction[]) {
```

Comment:
Can we mark internal functions as private or protected? It will help clarify the wrapper API, of which >95% of the usage will be:
```
executeTransaction
executeTransactions
executeMetaTransaction
signMetaTransaction
signMetaTransactions
```

Reply:
Or maybe the question should be, why make this function, as well as those below, public?
```
buildExecuteTransactionsWrapperTx
getMetaTransactionTypedData
buildMetaTxTypedData
```